File: explore_project.js
const fs = require('fs');
const path = require('path');

const projectRoot = __dirname;

// Fonction pour obtenir la structure du répertoire
function getDirectoryTree(dir, prefix = '', maxDepth = 20, currentDepth = 0) {
    if (currentDepth >= maxDepth) return '';
    
    let tree = '';
    try {
        const items = fs.readdirSync(dir);
        const sortedItems = items.sort((a, b) => {
            const isADir = fs.statSync(path.join(dir, a)).isDirectory();
            const isBDir = fs.statSync(path.join(dir, b)).isDirectory();
            if (isADir !== isBDir) return isBDir ? 1 : -1;
            return a.localeCompare(b);
        });

        sortedItems.forEach((item, index) => {
            const itemPath = path.join(dir, item);
            const isDir = fs.statSync(itemPath).isDirectory();
            const isLast = index === sortedItems.length - 1;
            
            // Skip node_modules, .git, etc.
            if (item.startsWith('.') || item === 'node_modules' || item === 'target') {
                return;
            }
            
            const connector = isLast ? '└── ' : '├── ';
            const nextPrefix = prefix + (isLast ? '    ' : '│   ');
            
            tree += prefix + connector + item + (isDir ? '/' : '') + '\n';
            
            if (isDir) {
                tree += getDirectoryTree(itemPath, nextPrefix, maxDepth, currentDepth + 1);
            }
        });
    } catch (err) {
        // Skip on error
    }
    
    return tree;
}

// Fonction pour trouver tous les fichiers .java
function findJavaFiles(dir, relativePath = '') {
    let javaFiles = [];
    try {
        const items = fs.readdirSync(dir);
        items.forEach(item => {
            const itemPath = path.join(dir, item);
            const relPath = path.join(relativePath, item);
            
            if (item.startsWith('.') || item === 'node_modules' || item === 'target') {
                return;
            }
            
            const stats = fs.statSync(itemPath);
            if (stats.isDirectory()) {
                javaFiles = javaFiles.concat(findJavaFiles(itemPath, relPath));
            } else if (item.endsWith('.java')) {
                javaFiles.push({
                    path: relPath.replace(/\\/g, '/'),
                    fullPath: itemPath
                });
            }
        });
    } catch (err) {
        // Skip on error
    }
    
    return javaFiles.sort((a, b) => a.path.localeCompare(b.path));
}

// Main execution
console.log('='.repeat(80));
console.log('EXPLORATION DU PROJET JAVA');
console.log('='.repeat(80));
console.log();

console.log('=== STRUCTURE DU PROJET ===');
console.log();
console.log(projectRoot + '/');
console.log(getDirectoryTree(projectRoot));
console.log();

const javaFiles = findJavaFiles(projectRoot);

console.log('=== FICHIERS JAVA TROUVÉS ===');
console.log();
console.log(`Total: ${javaFiles.length} fichiers .java`);
console.log();
javaFiles.forEach(file => {
    console.log(`- ${file.path}`);
});
console.log();

console.log('=== CONTENUS DES FICHIERS ===');
console.log();

javaFiles.forEach((file, index) => {
    console.log('='.repeat(80));
    console.log(`[${index + 1}/${javaFiles.length}] ${file.path}`);
    console.log('='.repeat(80));
    console.log();
    
    try {
        const content = fs.readFileSync(file.fullPath, 'utf-8');
        console.log(content);
    } catch (err) {
        console.log(`ERREUR: Impossible de lire le fichier - ${err.message}`);
    }
    
    console.log();
    console.log();
});

console.log('='.repeat(80));
console.log('EXPLORATION TERMINÉE');
console.log('='.repeat(80));

File: explore_project.py
#!/usr/bin/env python3
import os
import sys
from pathlib import Path

project_root = Path(__file__).parent

def get_directory_tree(directory, prefix="", max_depth=20, current_depth=0, ignore_patterns=None):
    """Generate a tree structure of the directory."""
    if ignore_patterns is None:
        ignore_patterns = {'.', '__pycache__', 'node_modules', 'target', '.git', '.vscode'}
    
    if current_depth >= max_depth:
        return ""
    
    tree = ""
    try:
        items = sorted(os.listdir(directory), key=lambda x: (not os.path.isdir(os.path.join(directory, x)), x))
        
        for i, item in enumerate(items):
            if item.startswith('.') or item in ignore_patterns:
                continue
                
            item_path = os.path.join(directory, item)
            is_dir = os.path.isdir(item_path)
            is_last = i == len(items) - 1
            
            connector = "└── " if is_last else "├── "
            next_prefix = prefix + ("    " if is_last else "│   ")
            
            tree += prefix + connector + item + ("/" if is_dir else "") + "\n"
            
            if is_dir:
                tree += get_directory_tree(item_path, next_prefix, max_depth, current_depth + 1, ignore_patterns)
    except PermissionError:
        pass
    
    return tree

def find_java_files(directory, relative_path=""):
    """Find all .java files in the directory."""
    java_files = []
    
    ignore_patterns = {'.', '__pycache__', 'node_modules', 'target', '.git', '.vscode'}
    
    try:
        for item in os.listdir(directory):
            if item.startswith('.') or item in ignore_patterns:
                continue
            
            item_path = os.path.join(directory, item)
            rel_path = os.path.join(relative_path, item)
            
            if os.path.isdir(item_path):
                java_files.extend(find_java_files(item_path, rel_path))
            elif item.endswith('.java'):
                java_files.append({
                    'path': rel_path.replace('\\', '/'),
                    'full_path': item_path
                })
    except PermissionError:
        pass
    
    return sorted(java_files, key=lambda x: x['path'])

# Main execution
print("=" * 80)
print("EXPLORATION DU PROJET JAVA")
print("=" * 80)
print()

print("=== STRUCTURE DU PROJET ===")
print()
print(str(project_root) + "/")
print(get_directory_tree(project_root))
print()

java_files = find_java_files(project_root)

print("=== FICHIERS JAVA TROUVÉS ===")
print()
print(f"Total: {len(java_files)} fichiers .java")
print()
for java_file in java_files:
    print(f"- {java_file['path']}")
print()

print("=== CONTENUS DES FICHIERS ===")
print()

for index, java_file in enumerate(java_files, 1):
    print("=" * 80)
    print(f"[{index}/{len(java_files)}] {java_file['path']}")
    print("=" * 80)
    print()
    
    try:
        with open(java_file['full_path'], 'r', encoding='utf-8') as f:
            print(f.read())
    except Exception as e:
        print(f"ERREUR: Impossible de lire le fichier - {e}")
    
    print()
    print()

print("=" * 80)
print("EXPLORATION TERMINÉE")
print("=" * 80)
